src/player_avatar.c

```diff
@@ -0,0 +1,731 @@
+#include "player_avatar.h"
+
+#include "global.h"
+
+#include "constants/global_fieldmap.h"
+#include "constants/sprites.h"
+
+#include "assert.h"
+#include "heap.h"
+#include "map_object.h"
+#include "overlay_00_thumb.h"
+#include "overlay_01.h"
+#include "overlay_01_021FE780.h"
+#include "overlay_01_02200858.h"
+#include "player_data.h"
+#include "unk_02034354.h"
+#include "unk_02035900.h"
+#include "unk_0205FD20.h"
+
+static PlayerAvatar *PlayerAvatar_Create(void);
+static void PlayerAvatar_Setup(PlayerAvatar *avatar, s32 state, u32 gender, PlayerSaveData *playerSaveData);
+static void PlayerAvatar_CreateMapObjectWithParams(PlayerAvatar *avatar, MapObjectManager *manager, u32 sprite, u32 direction, u32 x, u32 z);
+static LocalMapObject *PlayerAvatar_GetActiveMapObjectWithMovement1(MapObjectManager *manager);
+static LocalMapObject *PlayerAvatar_GetMapObjectConst(PlayerAvatar *avatar);
+static void PlayerAvatar_SetGender(PlayerAvatar *avatar, u32 gender);
+static void PlayerAvatar_SetFlagsBits(PlayerAvatar *avatar, PlayerAvatarFlags flags);
+static void PlayerAvatar_ClearFlagsBits(PlayerAvatar *avatar, PlayerAvatarFlags flags);
+static PlayerAvatarFlags PlayerAvatar_GetFlagsBitsMask(PlayerAvatar *avatar, PlayerAvatarFlags flags);
+static void PlayerAvatar_SetUnk28(PlayerAvatar *avatar, s32 param1);
+static void PlayerAvatar_SetUnk2C(PlayerAvatar *avatar, s32 param1);
+static void PlayerAvatar_SetPlayerSaveData(PlayerAvatar *avatar, PlayerSaveData *playerSaveData);
+static void PlayerSaveData_SetState(PlayerSaveData *playerSaveData, s32 state);
+static void PlayerAvatar_SetPlayerSaveDataState(PlayerAvatar *avatar, s32 state);
+
+u32 sub_0205C268(u32 param0) {
+    if (param0 < 100) {
+        return 255;
+    }
+    return ov00_021E6EBC();
+}
+
+u8 sub_0205C278(void) {
+    PlayerProfile *profile = sub_02034818(sub_0203769C() ^ 1);
+    GF_ASSERT(profile != NULL);
+    return PlayerProfile_GetVersion(profile);
+}
+
+u8 sub_0205C298(SaveData *saveData) {
+    if (PlayerProfile_GetVersion(Save_PlayerData_GetProfile(saveData)) == 0) {
```

Comment:
Version constant? We don't have one for 0 defined so I'm not sure what this is checking for

Reply:
I'm not sure what 0 even would be, so I don't know what to put for the constant